File: app/src/main/java/com/mayan/numbers/decimal/MainActivity.java
package com.mayan.numbers.decimal;


import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.mayan.numbers.decimal.calculator.DecimalToMayan;

import java.util.ArrayList;
import java.util.List;

/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 *
 */
public class MainActivity extends AppCompatActivity {
	
	private Button convertButton;
	private EditText convertText;
	
	protected void onCreate(Bundle savedInstanceState) {

		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		convertButton = (Button)findViewById(R.id.convertButton);
		convertText = (EditText)findViewById( R.id.convertText );
		convertButton.setOnClickListener( new View.OnClickListener() {
			public void onClick(View v) {
				convertButtonClicked(v);
			}
		});

		Bundle bundle = this.getIntent().getExtras();
		if(bundle != null && bundle.containsKey("errMsg")){
			int errMsg = bundle.getInt("errMsg");
			String errorMessage = getString(errMsg);
			this.showErrorAlert(errorMessage);
		}


	}

	protected void convertButtonClicked(View v) {
		try {
			String numberText = convertText.getText().toString();
			if( numberText.length() > 0 ){
				int decimalNumber = Integer.parseInt(numberText);
				if( decimalNumber > 0 ){
					DecimalToMayan decimalToMayan = new DecimalToMayan();
					List<Integer> numbers = decimalToMayan.getNumerics(decimalNumber);
					Intent intent = new Intent( this , DrawNumbersActivity.class );
					Bundle extras = new Bundle();
					int size = numbers.size();
					int[] dec = new int[size];
					for (int i = 0; i < size; i++) {
						dec[i] = numbers.get(i);
					}
					extras.putIntArray("numbers", dec);
					intent.putExtras(extras);
					startActivity(intent);					
				}else if( decimalNumber == 0 ){
                    List<Integer> numbers = new ArrayList<>();
                    numbers.add(0);
                    Intent intent = new Intent( this , DrawNumbersActivity.class );
                    Bundle extras = new Bundle();
                    int size = numbers.size();
                    int[] dec = new int[size];
                    for (int i = 0; i < size; i++) {
                        dec[i] = numbers.get(i);
                    }
                    extras.putIntArray("numbers", dec);
                    intent.putExtras(extras);
                    startActivity(intent);
                }
			}else{
				String message = getString(R.string.empty_text_convert);
				this.showErrorAlert(message);
			}
		} catch (Exception e) {
			String errorMessage = getString(R.string.error_when_converting_big_number);
			this.showErrorAlert(errorMessage);
			e.printStackTrace();
		}
	}
	
	
	private void showErrorAlert(String message){
		new AlertDialog.Builder( this )
				.setPositiveButton(getString(R.string.ok_message),null)
				.setMessage(message)
				.show();
	}

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_activity_menu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch(id){
            case R.id.main_activity_menu_item:
                exitApplication();
                break;
        }

        return super.onOptionsItemSelected(item);
    }

    private void exitApplication(){
        this.finishAffinity();
    }

}

File: app/src/main/java/com/mayan/numbers/decimal/MainActivity.kt
package com.mayan.numbers.decimal


import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle

import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.EditText

import androidx.appcompat.app.AppCompatActivity

import com.mayan.numbers.decimal.calculator.DecimalToMayan


/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 *
 */
 class MainActivity : AppCompatActivity() {

    private lateinit var convertButton: Button
    private lateinit var convertText: EditText

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        convertButton = findViewById(R.id.convertButton)
        convertText = findViewById(R.id.convertText)
        convertButton.setOnClickListener { _ ->
            convertButtonClicked()
        }

        val bundle = this@MainActivity.intent.extras
        if (bundle != null && bundle.containsKey("errMsg")) {
            val errMsg: Int = bundle.getInt("errMsg")
            val errorMessage: String = getString(errMsg)
            this@MainActivity.showErrorAlert(errorMessage)
        }


    }

    private fun convertButtonClicked() {
        try {
            val numberText = convertText.text.toString()
            if (numberText.isNotEmpty()) {
                val decimalNumber: Int = Integer.parseInt(numberText)
                if (decimalNumber > 0) {
                    val decimalToMayan = DecimalToMayan()
                    val numbers = decimalToMayan.getNumerics(decimalNumber)
                    val intent = Intent(this, DrawNumbersActivity::class.java)
                    val extras = Bundle()
                    val size = numbers.size
                    val dec: IntArray =  IntArray(size)
                    for ((i, number) in numbers.withIndex()) {
                        dec[i] = number
                    }
                    extras.putIntArray("numbers", dec)
                    intent.putExtras(extras)
                    startActivity(intent)
                } else if (decimalNumber == 0) {
                    val numbers = arrayListOf<Int>()
                    numbers.add(0)
                    val intent = Intent(this, DrawNumbersActivity::class.java)
                    val extras = Bundle()
                    val size = numbers.size
                    val dec = IntArray(size)
                    for ((i, number) in numbers.withIndex()) {
                        dec[i] = number
                    }
                    extras.putIntArray("numbers", dec)
                    intent.putExtras(extras)
                    startActivity(intent)
                }
            } else {
                val message = getString(R.string.empty_text_convert)
                this@MainActivity.showErrorAlert(message)
            }
        } catch (e: Exception) {
            val errorMessage = getString(R.string.error_when_converting_big_number)
            this@MainActivity.showErrorAlert(errorMessage)
            e.printStackTrace()
        }
    }


    private fun showErrorAlert(message: String) {
        AlertDialog.Builder(this)
            .setPositiveButton(getString(R.string.ok_message), null)
            .setMessage(message)
            .show()
    }


    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        this@MainActivity.menuInflater.inflate(R.menu.main_activity_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.main_activity_menu_item -> exitApplication()
            R.id.main_activity_decimal_item -> openDecimalConvert()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun exitApplication() {
        this@MainActivity.finishAffinity()
    }

    private fun openDecimalConvert() {
        val intent = Intent(this, ConvertToDecimalActivity::class.java)
        startActivity(intent)
    }


}
